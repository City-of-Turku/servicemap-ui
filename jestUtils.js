import React from 'react';
import { render } from '@testing-library/react';
import configureStore from 'redux-mock-store';
import { ThemeProvider } from '@mui/material/styles';
import { IntlProvider } from 'react-intl';
import { Provider } from 'react-redux';
import themes from './src/themes';
import finnish from './src/i18n/fi';

const mockStore = configureStore([]);

// Mock props for intl provider
const intlMock = {
  locale: 'fi',
  messages: finnish,
  wrapRichTextChunksInFragment: false,
};

const Providers = (mockState) => ({ children }) => {
  const store = mockStore(mockState);
  return (
    <Provider store={store}>
      <IntlProvider {...intlMock}>
        <ThemeProvider theme={themes.SMTheme}>
          {children}
        </ThemeProvider>
      </IntlProvider>
    </Provider>
  );
}

export const getRenderWithProviders = mockState => component => render(component, { wrapper: Providers(mockState) });