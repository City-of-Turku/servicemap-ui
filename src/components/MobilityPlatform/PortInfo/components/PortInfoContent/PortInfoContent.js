import { Typography } from '@mui/material';
import PropTypes from 'prop-types';
import React from 'react';
import { useIntl } from 'react-intl';
import { format } from 'date-fns';
import styled from '@emotion/styled';
import {
  StyledContainer, StyledFlexContainer, StyledHeaderContainer, StyledTextContainer,
} from '../../../styled/styled';

const PortInfoContent = ({ portItem, portCalls }) => {
  const intl = useIntl();

  const formatDateTime = dateTimeValue => format(new Date(dateTimeValue), 'dd.MM (HH:mm)');

  const renderArrivals = () => (
    <div>
      <StyledTextContainer>
        <Typography variant="subtitle1" component="h5">
          {intl.formatMessage({ id: 'mobilityPlatform.content.portInfo.arrivals' })}
        </Typography>
      </StyledTextContainer>
      {portCalls?.map(item => (
        <StyledFlexContainer>
          <FerryIcon color="rgba(7, 44, 115, 255)" className="icon-icon-hsl-ferry" />
          <StyledTextContainer>
            <Typography variant="body2" component="p">
              {item.vesselName}
            </Typography>
          </StyledTextContainer>
          <StyledTextContainer>
            <Typography variant="body2" component="p">
              {formatDateTime(item.portAreaDetails[0].eta)}
            </Typography>
          </StyledTextContainer>
        </StyledFlexContainer>
      ))}
    </div>
  );

  const renderDeparting = () => (
    <div>
      <StyledTextContainer>
        <Typography variant="subtitle1" component="h5">
          {intl.formatMessage({ id: 'mobilityPlatform.content.portInfo.departing' })}
        </Typography>
      </StyledTextContainer>
      {portCalls?.map(item => (
        <StyledFlexContainer>
          <FerryIcon color="rgba(7, 44, 115, 255)" className="icon-icon-hsl-ferry" />
          <StyledTextContainer>
            <Typography variant="body2" component="p">
              {item.vesselName}
            </Typography>
          </StyledTextContainer>
          <StyledTextContainer>
            <Typography variant="body2" component="p">
              {formatDateTime(item.portAreaDetails[0].etd)}
            </Typography>
          </StyledTextContainer>
        </StyledFlexContainer>
      ))}
    </div>
  );

  return (
    <StyledContainer>
      <StyledHeaderContainer>
        <Typography variant="subtitle1" component="h4">
          {portItem?.properties?.portAreaName}
        </Typography>
      </StyledHeaderContainer>
      <>
        {renderArrivals()}
        {renderDeparting()}
      </>
    </StyledContainer>
  );
};

const FerryIcon = styled.span(({ color }) => ({
  fontSize: 20,
  width: '20px',
  height: '20px',
  lineHeight: '21px',
  marginLeft: '6px',
  marginRight: '4px',
  marginTop: '8px',
  color,
}));

PortInfoContent.propTypes = {
  portItem: PropTypes.shape({
    properties: PropTypes.shape({
      portAreaName: PropTypes.string,
    }),
  }),
  portCalls: PropTypes.arrayOf(PropTypes.shape({
    vesselName: PropTypes.string,
  })),
};

PortInfoContent.defaultProps = {
  portItem: {},
  portCalls: [],
};

export default PortInfoContent;
