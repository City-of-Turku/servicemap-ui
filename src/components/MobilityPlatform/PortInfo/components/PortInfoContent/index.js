import PortInfoContent from './PortInfoContent';

export default PortInfoContent;
