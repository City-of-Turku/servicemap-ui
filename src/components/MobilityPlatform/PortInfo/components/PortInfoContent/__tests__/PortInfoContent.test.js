// Link.react.test.js
import React from 'react';
import { getRenderWithProviders } from '../../../../../../../jestUtils';
import finnishTranslations from '../../../../../../i18n/fi';
import PortInfoContent from '../index';

const mockProps = {
  portItem: {
    properties: {
      portAreaName: 'Testisatama',
    },
  },

  portCalls: [
    {
      portCallId: 12340,
      portCallTimestamp: '2024-06-06T05:00:00z',
      vesselName: 'Testilaiva',
      portAreaDetails: [
        {
          eta: '2024-06-06T04:35:00z',
          etd: '2024-06-06T05:55:00z',
        },
      ],
    },
  ],
};

const renderWithProviders = getRenderWithProviders({});

describe('<PortInfoContent />', () => {
  it('should match snapshot', () => {
    const { container } = renderWithProviders(<PortInfoContent {...mockProps} />);
    expect(container).toMatchSnapshot();
  });

  it('does show text', () => {
    const { container } = renderWithProviders(<PortInfoContent {...mockProps} />);
    const p = container.querySelectorAll('p');
    const h4 = container.querySelector('h4');
    const h5 = container.querySelectorAll('h5');
    expect(h4.textContent).toContain(mockProps.portItem.properties.portAreaName);
    expect(h5[0].textContent).toContain(finnishTranslations['mobilityPlatform.content.portInfo.arrivals']);
    expect(h5[1].textContent).toContain(finnishTranslations['mobilityPlatform.content.portInfo.departing']);
    expect(p[0].textContent).toContain(mockProps.portCalls[0].vesselName);
    expect(p[1]).toBeInTheDocument();
    expect(p[3]).toBeInTheDocument();
  });

  it('does show span elements', () => {
    const { container } = renderWithProviders(<PortInfoContent {...mockProps} />);
    const span = container.querySelectorAll('span');
    expect(span[0]).toBeInTheDocument();
    expect(span[1]).toBeInTheDocument();
  });
});
