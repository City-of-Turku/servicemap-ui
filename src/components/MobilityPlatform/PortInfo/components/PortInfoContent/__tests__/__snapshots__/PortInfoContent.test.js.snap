// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PortInfoContent /> should match snapshot 1`] = `
<div>
  <div
    class="css-1ynyhby"
  >
    <div
      class="css-1dznbe2"
    >
      <h4
        class="MuiTypography-root MuiTypography-subtitle1 css-dlbwt1-MuiTypography-root"
      >
        Testisatama
      </h4>
    </div>
    <div>
      <div
        class="css-1fhgjcy"
      >
        <h5
          class="MuiTypography-root MuiTypography-subtitle1 css-dlbwt1-MuiTypography-root"
        >
          Saapuvat laivat:
        </h5>
      </div>
      <div
        class="css-1do907m"
      >
        <span
          class="icon-icon-hsl-ferry css-o38fol"
          color="rgba(7, 44, 115, 255)"
        />
        <div
          class="css-1fhgjcy"
        >
          <p
            class="MuiTypography-root MuiTypography-body2 css-1ak20dk-MuiTypography-root"
          >
            Testilaiva
          </p>
        </div>
        <div
          class="css-1fhgjcy"
        >
          <p
            class="MuiTypography-root MuiTypography-body2 css-1ak20dk-MuiTypography-root"
          >
            06.06 (07:35)
          </p>
        </div>
      </div>
    </div>
    <div>
      <div
        class="css-1fhgjcy"
      >
        <h5
          class="MuiTypography-root MuiTypography-subtitle1 css-dlbwt1-MuiTypography-root"
        >
          Lähtevät laivat:
        </h5>
      </div>
      <div
        class="css-1do907m"
      >
        <span
          class="icon-icon-hsl-ferry css-o38fol"
          color="rgba(7, 44, 115, 255)"
        />
        <div
          class="css-1fhgjcy"
        >
          <p
            class="MuiTypography-root MuiTypography-body2 css-1ak20dk-MuiTypography-root"
          >
            Testilaiva
          </p>
        </div>
        <div
          class="css-1fhgjcy"
        >
          <p
            class="MuiTypography-root MuiTypography-body2 css-1ak20dk-MuiTypography-root"
          >
            06.06 (08:55)
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
